stablehlo/reference/Ops.cpp

```diff
@@ -321,6 +321,14 @@ SmallVector<InterpreterValue> eval(Region &region,
   scope.add(block.getArguments(), args);
 
   for (Operation &operation : block) {
+    if (!llvm::all_of(operation.getResults(), [](OpResult r) {
```

Comment:
Thanks for taking care of this change.
I was reconsidering adding this check vs per op check because there are only 6 dynamic_ops and we are going to update the reference interpreter to support dynamic reshape. If we end up updating  interpreter to allow dynamic shape one op by one, then this check needs to be relaxed per op basis.

For now, we can keep the check, no change needed here.

Reply:
This is relevant for all ops, though. All ops can have dynamic results (except for e.g. `constant`). Supporting dynamic types in the interpreter is a global change so IMO it makes sense to have a global check.